packages/block-library/src/navigation/index.php

```diff
@@ -253,8 +253,15 @@ function render_block_core_navigation( $attributes, $content, $block ) {
 
 	$inner_blocks = $block->inner_blocks;
 
-	// If `__unstableLocation` is defined, create inner blocks from the classic menu assigned to that location.
-	if ( empty( $inner_blocks ) && array_key_exists( '__unstableLocation', $attributes ) ) {
+	// If `__unstableLocation` is defined and:
+	// - we don't have a relationship to a `wp_navigation` Post (via `navigationMenuId`).
+	// - we don't have any inner blocks (e.g. a block pattern).
+	// ...then create inner blocks from the classic menu assigned to that location.
+	if (
+		array_key_exists( '__unstableLocation', $attributes ) &&
+		! array_key_exists( 'navigationMenuId', $attributes ) &&
+		empty( $inner_blocks )
```

Comment:
I think if we remove this line then we'll also solve this issue: https://github.com/WordPress/gutenberg/issues/36865

Reply:
So to be absoltutely clear, the behaviour priority order should be:

1. `navigationMenuId`
2. `__unstableLocation`
3. `innerBlocks`

I.e. if there is menuId then ignore unstable location. But if not then use unstable location. If that doesn't exist then fallback to any inner blocks. But in all cases we should ignore inner blocks if the other two are defined.

Reply:
So removing `empty( $inner_blocks )` will fix the priority and there the Issue you reference https://github.com/WordPress/gutenberg/issues/36865

Reply:
Though if there's no menu at the location it should at least fall back to the inner blocks again.

Reply:
Like this?

https://github.com/WordPress/gutenberg/pull/36863/files#r757345433